examples/userapi/00_dsl.ipynb

```diff
@@ -0,0 +1,1033 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# The Devito domain specific language: an overview\n",
+    "\n",
+    "This notebook presents an overview of the symbolic layers of Devito to express and discretize operators, in particular partial differential equations (PDEs).\n",
+    "\n",
+    "For simplicity, let's import all Devito modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from devito import *"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## From equations to code in a few lines of Python\n",
+    "\n",
+    "Today's objective is to demonstrate how Devito and its [SymPy](http://www.sympy.org/en/index.html)-powered symbolic API can be used to solve partial differential equations using the finite difference method with highly optimized stencils in a few lines of Python. We will show how to derive computational stencils directly from the equation in an automated fashion and how we can use Devito to generate and execute optimized C code at runtime to solve our problem.\n",
+    "\n",
+    "\n",
+    "## Defining the physical domain \n",
+    "Before we can start creating stencils we will need to give Devito a few details about the computational domain in which we want to solve our problem. For this purpose we create a `Grid` object that stores the physical `extent` (the size) of our domain and knows how many points we want to use in each dimension to discretize our data.\n",
+    "\n",
+    "<img src=\"figures/grid.png\" style=\"width: 220px;\"/>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Grid[extent=(1.0, 1.0), shape=(5, 6), dimensions=(x, y)]"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "grid = Grid(shape=(5, 6), extent=(1., 1.))\n",
+    "grid"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Functions and data\n",
+    "\n",
+    "To express our equation in symbolic form and discretize it using finite differences, Devito provides a set of `Function` types. A `Function` object created from these does two things:\n",
```

Comment:
Possibly shorten "A Function object created from these does two things" to "A Function object does the following"? What "these" is referring to is slightly ambiguous when you read through the notebook, and it doesn't really change the meaning to remove it.

Reply:
I made it even shorter, see update

Good spot. As I said, most of this notebook is copy-pasted from the old SC 17 notebook, and I'm counting on people to spot stuff like these ;)